pkg/logs/internal/decoder/auto_multiline_detection/tokenizer.go

```diff
@@ -376,3 +376,21 @@ func tokensToString(tokens []Token) string {
 	}
 	return str
 }
+
+// isMatch compares two sets of tokens and returns true if they match.
```

Comment:
Set commonly means an unordered collection of distinct elements (take the mathematical definition of  a set or the python `set` type), and the comparison operation would reflect that. Since none of these two properties is true for the arguments of this function, I would suggest not calling its arguments "sets".

```suggestion
// isMatch compares two sequences of tokens and returns true if they match.
```